soco/events_asyncio.py

```diff
@@ -0,0 +1,582 @@
+# pylint: disable=not-context-manager,import-error,wrong-import-position
+
+# NOTE: The pylint not-content-manager warning is disabled pending the fix of
+# a bug in pylint. See https://github.com/PyCQA/pylint/issues/782
+
+# Disable while we have Python 2.x compatability
+# pylint: disable=useless-object-inheritance
+
+
+"""Classes to handle Sonos UPnP Events and Subscriptions.
+
+The `Subscription` class from this module will be used in
+:py:mod:`soco.services` if `config.EVENTS_MODULE` is set
+to point to this module.
+
+Example:
+
+    Run this code, and change your volume, tracks etc::
+
+        from __future__ import print_function
+        import logging
+
+        logging.basicConfig()
+        import soco
+        import asyncio
+        from pprint import pprint
+
+        from soco import events_asyncio
+
+        soco.config.EVENTS_MODULE = events_asyncio
+
+
+        def print_event(event):
+            try:
+                pprint(event.variables)
+            except Exception as e:
+                pprint("There was an error in print_event:", e)
+
+
+        def _get_device():
+            device = soco.discover().pop().group.coordinator
+            print(device.player_name)
+            return device
+
+
+        async def main():
+            # pick a device at random and use it to get
+            # the group coordinator
+            loop = asyncio.get_event_loop()
+            device = await loop.run_in_executor(None, _get_device)
+            sub = await device.renderingControl.subscribe()
+            sub2 = await device.avTransport.subscribe()
+            sub.callback = print_event
+            sub2.callback = print_event
+
+            async def before_shutdown():
+                await sub.unsubscribe()
+                await sub2.unsubscribe()
+                await events_asyncio.event_listener.async_stop()
+
+            await asyncio.sleep(100)
+            await before_shutdown()
+
+
+        if __name__ == "__main__":
+            asyncio.run(main())
+
+"""
+
```

Comment:
Where does this appear in the documentation tree?

Reply:
I made sure its included with https://github.com/SoCo/SoCo/pull/815/commits/2a1f7369992f2d535777c65c7213db9f2cd99401